packages/block-library/src/html/edit.js

```diff
@@ -73,11 +73,16 @@ export default function HTMLEdit( { attributes, setAttributes, isSelected } ) {
 					</VisuallyHidden>
 				</>
 			) : (
-				<PlainText
-					value={ attributes.content }
-					onChange={ ( content ) => setAttributes( { content } ) }
-					placeholder={ __( 'Write HTML…' ) }
-					aria-label={ __( 'HTML' ) }
+				<EditorView
+					editorId={'block-library-html__editor'}
+					editorInstructionsText={__(
+						`This editor allows you to input your custom HTML.`
+					)}
+					initialConfig={{
+						content: attributes.content,
+						onChange,
+						mode: 'html',
+					}}
```

Comment:
I have a concern here: ideally we don't want to display editor UI inside the canvas, only content. Is there any way to do that with CodeMirror? So stuff like Autocomplete etc. should ideally be rendered outside the iframe so it can be styled as UI. I don't know if that would mean initialising outside the iframe and targeting a node in the canvas, or initialising inside the the iframe and setting a different area for UI.

Reply:
Another concern is that I'd expect however the code block is styled/syntax highlighted to be the same on the front-end. Is that possible?

Reply:
I guess I should read better: I just saw that this is for the HTML block and not the Code block, so these comments are not relevant. :)